packages/notifications/push/amplify_push_notifications/ios/Classes/AmplifyPushNotificationsPlugin.swift

```diff
@@ -1,19 +1,183 @@
+
+        if permissions["alert"] as? Bool == true {
+            options.insert(.alert)
+        }
+
+        if permissions["badge"] as? Bool == true {
+            options.insert(.badge)
+        }
+
+        if permissions["sound"] as? Bool == true {
+            options.insert(.sound)
+        }
+
+        if permissions["criticalAlert"] as? Bool == true {
+            options.insert(.criticalAlert)
+        }
+
+        if permissions["provisional"] as? Bool == true {
+            options.insert(.provisional)
+        }
+
+        UNUserNotificationCenter.current().requestAuthorization(options: options) { granted, error in
+            if (error != nil) {
+                completionHandler(.failure(
+                    FlutterError(code: "RequsetPermissionsError",
+                                 message: "Error occurred requesting notitication center authorization.",
+                                 details: error?.localizedDescription)))
+            } else {
+                completionHandler(.success(granted))
+            }
+        }
+    }
+
+    private func getPermissionStatus(completionHandler: @escaping (String) -> Void) {
+        UNUserNotificationCenter.current().getNotificationSettings { settings in
+            var status: String
+
+            switch settings.authorizationStatus {
+            case .notDetermined:
+                status = "NotDetermined"
+            case .denied:
+                status = "Denied"
+            case .authorized:
+                status = "Authorized"
+            case .ephemeral:
+                status = "Ephemeral"
+            case .provisional:
+                status = "Provisional"
+            @unknown default:
+                status = "NotDetermined"
+            }
+            completionHandler(status)
+        }
+    }
+
+    private func getLaunchNotification() -> Any? {
+        let launchNotification = self.launchNotification
+        self.launchNotification = nil
+        return launchNotification
+    }
+
+    private func getBadgeCount() -> Int {
+        return UIApplication.shared.applicationIconBadgeNumber
+    }
+
+    private func setBadgeCount(
+        args: Any?,
+        completionHandler: @escaping (PushNotificationOperationResult<Bool>) -> Void
+    ) {
+        guard let count = args as? Int else {
+            completionHandler(.failure(invalidArgumentsError))
+            return
+        }
+
+        UIApplication.shared.applicationIconBadgeNumber = count
+        completionHandler(.success(true))
+    }
+
+    private func completeNotification(
+        args: Any?,
+        completionHandler: @escaping (PushNotificationOperationResult<Bool>) -> Void
+    ) {
+        guard let completionHandlerId = args as? String else {
+            completionHandler(.failure(invalidArgumentsError))
+            return
+        }
+
+        if let completionHanlder = remoteNotificationCompletionHandlers[completionHandlerId] {
```

Comment:
```suggestion
        if let completionHandler = remoteNotificationCompletionHandlers[completionHandlerId] {
```

Reply:
Actually can you call it something else so it's different than the method's completion handler?